app/models/application.rb

```diff
@@ -149,4 +136,12 @@ def sponsor_pick?
       value.to_s != '0' ? flags.concat([flag.to_s]).uniq : flags.delete(flag.to_s)
     end
   end
+
+  def project1
+    Project.find_by(id: application_data['project1_id'])
```

Comment:
I would say as a data object first. if we have the data object, moving it to the database as persisted is already half way there. since we can keep the same abstraction. I will work on a proposal for this today, to better show what I mean :)